tools/rostopic/src/rostopic/__init__.py

```diff
@@ -297,7 +297,7 @@ def msgevalgen(pattern):
     for f in fields:
         if '[' in f:
             field_name, rest = f.split('[', 1)
-            if rest[-1] != ']':
+            if not rest or rest[-1] != ']':
                 print("missing closing ']' in slice spec '%s'" % f, file=sys.stderr)
```

Comment:
Just using `not rest.endswith(']')` seems like a good suggestion. Then the second check is also not necessary.

The additional tests would be nice but the important part is that they are already captured.